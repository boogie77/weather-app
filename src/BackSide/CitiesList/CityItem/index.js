import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

export default class CityItem extends Component {
  static propTypes = {
    onSelect: PropTypes.func,
    isSelected: PropTypes.bool,
    city: PropTypes.shape({
      title: PropTypes.string.isRequired
    })
  };

  onClick = () => {
    const { onSelect, city } = this.props;
    onSelect(city);
  };

  render() {
    const { city, isSelected } = this.props;
    return (
      <li
        className={classNames("list-item", { "is-selected": isSelected })}
        onClick={this.onClick}
      >
        {city.title}
      </li>
    );
  }
}
