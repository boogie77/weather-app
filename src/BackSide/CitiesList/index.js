import React from "react";
import PropTypes from "prop-types";
import { CityItem } from "./CityItem";
import "./list.css";

const CitiesList = ({ onSelect, cities, currentCity }) => {
  return (
    <ul className="list">
      {cities.map(city => (
        <CityItem
          key={city.title}
          isSelected={city.title === currentCity.title}
          onSelect={onSelect}
          city={city}
        />
      ))}
    </ul>
  );
};

CitiesList.propTypes = {
  onSelect: PropTypes.func,
  cities: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired
    })
  ),
  currentCity: PropTypes.shape({
    title: PropTypes.string.isRequired
  })
};

export default CitiesList;
