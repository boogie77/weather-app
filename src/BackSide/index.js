import React, { Component } from "react";
import PropTypes from "prop-types";
import CitiesList from "./CitiesList";
import "./button.css";

export default class BackSide extends Component {
  static propTypes = {
    cities: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired
      })
    ),
    onChange: PropTypes.func,
    onSave: PropTypes.func,
    currentCity: PropTypes.shape({
      title: PropTypes.string.isRequired
    })
  };

  render() {
    const { cities, onChange, onSave, currentCity } = this.props;
    return (
      <div className="card-back">
        <CitiesList
          cities={cities}
          onSelect={onChange}
          currentCity={currentCity}
        />
        <button className="button" onClick={this.props.onSave}>
          Apply
        </button>
      </div>
    );
  }
}
