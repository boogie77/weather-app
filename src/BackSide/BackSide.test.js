import React from "react";
import BackSide from "./index";
import CitiesList from "./CitiesList";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

const mockCities = [{ title: "City 1" }, { title: "City 2" }];
const mockOnSave = jest.fn();
const mockOnChange = jest.fn();

describe("<BackSide />", () => {
  it("renders and matches the snapshot", () => {
    const wrapper = shallow(
      <BackSide cities={mockCities} currentCity={mockCities[0]} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("event handlers", () => {
    const wrapper = shallow(
      <BackSide
        onSave={mockOnSave}
        onChange={mockOnChange}
        cities={mockCities}
        currentCity={mockCities[0]}
      />
    );

    it("calls the provided onChange callback when CitiesList onSelect triggered", () => {
      wrapper
        .find(CitiesList)
        .last()
        .prop("onSelect")();
      expect(mockOnChange).toHaveBeenCalled();
    });

    it("calls the provided onSave callback when save button clicked", () => {
      wrapper
        .find("button")
        .last()
        .simulate("click");
      expect(mockOnSave).toHaveBeenCalled();
    });
  });
});
