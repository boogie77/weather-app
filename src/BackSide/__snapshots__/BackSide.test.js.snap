// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackSide /> renders and matches the snapshot 1`] = `
<div
  className="card-back"
>
  <CitiesList
    cities={
      Array [
        Object {
          "title": "City 1",
        },
        Object {
          "title": "City 2",
        },
      ]
    }
    currentCity={
      Object {
        "title": "City 1",
      }
    }
  />
  <button
    className="button"
  >
    Apply
  </button>
</div>
`;
